modules/post/multi/manage/getty.rb

```diff
@@ -0,0 +1,96 @@
+
+class MetasploitModule < Msf::Post
+
+  def initialize(info={})
+    super( update_info( info,
+        'Name'          => 'Multiple Linux / Unix Post TTY Shell',
+        'Description'   => %q{
+          This module attempts to acquire a TTY via python or expect.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        => ['RageLtMan <rageltman[at]sempervictus>'],
+        'Platform'      => [ 'linux','unix','osx','solaris','aix','bsd' ],
+        'References'    =>
+          [
+            [ 'URL', 'http://pentestmonkey.net/blog/post-exploitation-without-a-tty']
+          ],
+        'SessionTypes'  => [ 'shell' ]
+      ))
+
+      register_options([
+        OptBool.new('USE_EXPECT', [true, 'Try expect shell', false]),
+        OptBool.new('USE_PYTHON', [true, 'Try python shell', true]),
+        OptString.new('TTY_SHELL', [true, 'Shell to spawn in TTY', '/bin/sh'])
+      ], self.class)
+
+  end
+
+  def run
+    # Error checking the results of a TTY shell attempt is tricky.
+    # After the TTY is acquired, cmd_exec no longer work until the next run
+    if have_tty?
+      print_good("Already have TTY")
+      return
+    end
+    if check_expect?
+      # This drops a file to disk, but works more often intesting
+      vprint_good("Using expect method")
+      expect_tty
+    elsif check_python?
+      # First choice since it's in-memory
+      vprint_good("Using Python method")
+      # Given that the python binary keeps running we get no output.
+      cmd_exec("python -c \"import pty;pty.spawn('#{datastore['TTY_SHELL']}')\""), nil, 2)
```

Comment:
```suggestion
      cmd_exec("python -c \"import pty;pty.spawn('#{datastore['TTY_SHELL']}')\"", nil, 2)
```

Reply:
Thanks @bcoles - i had removed and reinstated that `nil,2` thing in my attempts to make it more consistent. Dont code tired kids :).